medicines/doc-index-updater/src/pars_upload.rs

```diff
@@ -63,15 +72,22 @@ async fn queue_pars_upload(
     job_id: Uuid,
     form_data: FormData,
     state_manager: impl JobStatusClient,
-) -> Result<(), SubmissionError> {
-    let (metadata, file_data) = read_pars_upload(form_data).await.map_err(|e| {
+) -> Result<(), Rejection> {
+    let (metadatas, file_data) = read_pars_upload(form_data).await.map_err(|e| {
         tracing::debug!("Error reading PARS upload: {:?}", e);
-        e
+        warp::reject::custom(e)
     })?;
-    let storage_file = add_file_to_temporary_blob_storage(job_id, file_data).await?;
-    let document = document_from_form_data(storage_file, metadata);
 
-    let _ = check_in_document_handler(document, state_manager).await;
+    for metadata in metadatas {
+        let storage_file = add_file_to_temporary_blob_storage(job_id, &file_data)
+            .await
+            .map_err(warp::reject::custom)?;
+
+        let document = document_from_form_data(storage_file, metadata);
+
+        check_in_document_handler(document, &state_manager).await?;
```

Comment:
There's a problem here in that we've got multiple documents on the queue with the same job_id. It might be better to have an array of job ids, one for each product, so they can succeed or fail independently. Otherwise, we'd need to have an array of statuses per job id, or some other more complicated state management.

We're also not doing anything with the `JobStatusResponse` here and maybe we should.